Apps/WebClient/src/ClientApp/app/router.ts

```diff
@@ -96,17 +98,31 @@ router.beforeEach(async (to, from, next) => {
       }
     });
   } else {
-    next();
+    let userIsAuthenticated: boolean =
+      store.getters["auth/oidcIsAuthenticated"];
+    let userIsRegistered: boolean = store.getters["user/userIsRegistered"];
+
+    // If the user is authenticated but not registered, the registration must be completed
+    if (
+      userIsAuthenticated &&
+      !userIsRegistered &&
+      !to.meta.routeIsOidcCallback &&
+      !to.path.startsWith("/logout")
+    ) {
+      next({ path: REGISTRATION_PATH });
```

Comment:
we need to make sure we are passing any parameters to registration path if there is any

Reply:
At this point there cannot be any parameters since the user is attempting to go to an authenticated site